File: Java/LeetCode/Number of Different Integers in a String.java
// https://leetcode.com/problems/number-of-different-integers-in-a-string/

import java.math.BigInteger;

class Solution {
    
    public int numDifferentIntegers(String word) {
        String[] nums = word.split("\\D+"); // or "[a-z]+"
        Set<BigInteger> uniqueNums = new HashSet<>();
        
        for (String num : nums) {
            if (!num.isEmpty()) {
                uniqueNums.add(new BigInteger(num));
            }
        }

        return uniqueNums.size();
    }

}

File: Kotlin/LeetCode/Number of Different Integers in a String.kt
// https://leetcode.com/problems/number-of-different-integers-in-a-string/

class Solution {

    fun numDifferentIntegers(word: String): Int {
        val nums = word.split("\\D+".toRegex()) // or "[a-z]+"
        val uniqueNums = mutableSetOf<BigInteger>()

        nums.forEach {
            if (!it.isEmpty()) {
                uniqueNums.add(BigInteger(it))
            }
        }

        return uniqueNums.size
    }

}
